junit-platform-commons/src/main/java/org/junit/platform/commons/util/PackageUtils.java

```diff
@@ -62,12 +63,38 @@ private PackageUtils() {
 		Preconditions.notNull(function, "function must not be null");
 		Package typePackage = type.getPackage();
 		if (typePackage != null) {
-			String value = function.apply(typePackage);
-			if (value != null) {
-				return Optional.of(value);
-			}
+			return Optional.ofNullable(function.apply(typePackage));
 		}
 		return Optional.empty();
 	}
 
+	/**
+	 * Get the value of the specified attribute name, specified as a string,
+	 * or an empty {@link Optional} if the attribute was not found. The attribute
+	 * name is case-insensitive.
+	 *
+	 * <p>This method also returns an empty {@link Optional} value holder
+	 * if any exception is caught while loading the manifest file via the
+	 * specified class loader.
+	 *
+	 * @param loader the {@link ClassLoader} used to load the manifest with
+	 * @param name the attribute name as a string
+	 * @return an {@code Optional} containing the attribute value; never
+	 * {@code null} but potentially empty
+	 * @throws PreconditionViolationException if the supplied loader is
+	 * {@code null} or the specified name is blank
+	 * @see Manifest
+	 */
+	public static Optional<String> getAttribute(ClassLoader loader, String name) {
+		Preconditions.notNull(loader, "loader must not be null");
+		Preconditions.notBlank(name, "name must not be blank");
+		try {
+			Manifest manifest = new Manifest(loader.getResourceAsStream("META-INF/MANIFEST.MF"));
```

Comment:
I think I found the reason it doesn't work for me. This line reads the first manifest it finds on the classpath. Which, in my case, is a JAR in the JDK. The same thing (with a different JAR) happened when I tried it on a Docker container (openjdk:8). So it's not just a macOS problem.

I'm working on a fix now.